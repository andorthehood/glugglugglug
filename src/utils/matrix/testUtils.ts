export function roundValue(value: number, precision = 6) {
	const factor = 10 ** precision;
	const rounded = Math.round(value * factor) / factor;
	return Object.is(rounded, -0) ? 0 : rounded;
}

export function formatMatrix(values: number[]) {
	return Array.from({ length: 4 }, (_, row) => {
		const start = row * 4;
		return values.slice(start, start + 4).map((value) => roundValue(value));
	});
}

export function formatVector(values: number[]) {
	return values.map((value) => roundValue(value));
}
