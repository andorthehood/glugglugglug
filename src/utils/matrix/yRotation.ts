export function yRotation(angleInRadians: number) {
	const c = Math.cos(angleInRadians);
	const s = Math.sin(angleInRadians);

	return [c, 0, -s, 0, 0, 1, 0, 0, s, 0, c, 0, 0, 0, 0, 1];
}
