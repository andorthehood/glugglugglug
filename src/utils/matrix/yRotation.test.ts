import { describe, expect, it } from "vitest";
import { yRotation } from "./yRotation";
import { formatMatrix } from "./testUtils";

describe("yRotation", () => {
	it("creates a rotation matrix around the y axis", () => {
		const angle = Math.PI / 4;

		expect(formatMatrix(yRotation(angle))).toMatchSnapshot();
	});
});
