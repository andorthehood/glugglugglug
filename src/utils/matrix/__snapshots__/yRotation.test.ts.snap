// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`yRotation > creates a rotation matrix around the y axis 1`] = `
[
  [
    0.707107,
    0,
    -0.707107,
    0,
  ],
  [
    0,
    1,
    0,
    0,
  ],
  [
    0.707107,
    0,
    0.707107,
    0,
  ],
  [
    0,
    0,
    0,
    1,
  ],
]
`;
